### DISCUSSION TITLE
Scoppy and the Pico 2 (and other RP235x boards).

### DISCUSSION THREAD

**Discussion by fhdm-dev** [fhdm-dev/scoppy#153] [General]
Unfortunately the Scoppy firmware will be affected by the [E9-Erratum](https://hackaday.com/2024/09/04/the-worsening-raspberry-pi-rp2350-e9-erratum-situation/) issue. 

Because of this I don't know if or when there will be a version of Scoppy for the Pico 2.

**Comment by kd7eir** [comment]
Let's hope for a quick Pico 2A?

**Comment by shtirlic** [comment]
Hi, is it correct that logic analyzer gpios and everything without pull down is not affected? Hoping to see the firmware build  for 2350.

**Reply by fhdm-dev** [reply]
I believe so. It's just the voltage range GPIO that are affected (when used as inputs).

**Comment by shtirlic** [comment]
@fhdm-dev Looking forward for beta release for 2350, want to use it with my Pixel

**Comment by Seba-SL** [comment]
Hi, is for this that the uf2 file dont work in my raspberry pi pico2 ?

**Reply by fhdm-dev** [reply]
It's because the uf2 that is currently available is specifically for the original Pico. I will need to create a uf2 for the Pico 2 (or a universal uf2 that will work with both).

**Reply by ohault** [reply]
> It's because the uf2 that is currently available is specifically for the original Pico. I will need to create a uf2 for the Pico 2 (or a universal uf2 that will work with both).

Hi, it will be highly appreciate for both the Pico 2 and Pico 2 W. Thank you.

**Comment by jhonas4565** [comment]
> Lamentablemente, el firmware de Sppy se verá afectado por el problema [E9-Erratum](https://hackaday.com/2024/09/04/the-worsening-raspberry-pi-rp2350-e9-erratum-situation/) .
> 
> Por eso no sé si habrá una versión de Scoppy para Pico 2 o cuándo.

Still nothing for the 2w peak?? I had bought the raspberry without reading the forum.

**Reply by p3t3rm0v** [reply]
Not solved with new s4?

**Reply by fhdm-dev** [reply]
Hi. It was solved a while ago. Scoppy firmware for the Pico 2 and Pico 2 W can be downloaded from  [here](https://oscilloscope.fhdm.xyz/wiki/firmware-versions) and should work with the A4 stepping (though I haven't tested it).

**Comment by fhdm-dev** [comment]
Scoppy firmware for the Pico 2 is now available. It can be downloaded from [here](https://oscilloscope.fhdm.xyz/wiki/firmware-versions).

Firmware for the Pico 2 W will (hopefully) be available soon.

**Comment by fhdm-dev** [comment]
Scoppy firmware for the Pico 2 W is now available. It can be downloaded from [here](https://oscilloscope.fhdm.xyz/wiki/firmware-versions).

**Reply by ohault** [reply]
> Scoppy firmware for the Pico 2 W is now available. It can be downloaded from [here](https://oscilloscope.fhdm.xyz/wiki/firmware-versions).

Thank you @fhdm-dev. I have just successfully tested with a Pico 2 W.

**Reply by sergio-caracas** [reply]
Now that the erratum is fixed will you consider s new version compatible with RP2350 A4?

**Reply by fhdm-dev** [reply]
Hi. 

The existing Pico 2 firmware should work with the RP2350 A4 stepping.

I don't have an A4 stepping that I can test with but according to [this](https://forums.raspberrypi.com/viewtopic.php?p=2329181&hilit=a4#p2329093) thread on the Rasperry Pi Forums, firmware written for the A2 stepping is compatible with the A4.
